src/WebSockets.jl

```diff
@@ -248,11 +252,13 @@ is_control_frame(msg::WebSocketFragment) = (msg.opcode & 0b0000_1000) > 0
 
 """ Respond to pings, ignore pongs, respond to close."""
 function handle_control_frame(ws::WebSocket,wsf::WebSocketFragment)
+
     if wsf.opcode == OPCODE_CLOSE
         # Reply with an empty CLOSE frame
         locked_write(ws.socket, true, "", OPCODE_CLOSE)
         ws.is_closed = true
-        wait(ws.socket.closenotify)
+        close(ws.socket)
```

Comment:
Okay, I'd missed this `close` here. If this also doesn't work, then that's a Julia base bug.